### DISCUSSION TITLE
Issues with   import(`../messages/${locale}.json`))

### DISCUSSION THREAD

**Discussion by Answere1** [amannn/next-intl#1103] [Q&A]
I have a trouble with write code. In your project (examples/example-app-router/src/i18n.ts) in return you wrote like `import('../messages/${locale}.json')`. I don't know why, but next says no such module is found. 

Although I have the correct folder and file structure. However, if I write `import('../messages/en.json')` or `import('../messages/de.json')`(desired language) then everything is fine.  

Having said that, I have 4 languages and I can't give the exact path either, writing 4 imports it seems wrong

**Comment by meJohnnyRobert** [comment]
This is how I have mine configured and it works:

```js
// src/i18n.ts

import { notFound } from 'next/navigation';
import { getRequestConfig } from 'next-intl/server';

// Can be imported from a shared config
const locales = ['en', 'ro'];

export default getRequestConfig(async ({ locale }) => {
  // Validate that the incoming `locale` parameter is valid
  if (!locales.includes(locale as any)) notFound();

  // Updated path to reflect the new location at src/messages
  return {
    messages: (await import(`./messages/${locale}.json`)).default
  };
});
```

I hope it helps.

**Reply by Answere1** [reply]
![Снимок экрана 2024-06-03 205658](https://github.com/amannn/next-intl/assets/153187097/9ef9e66c-106b-4abc-8054-6e655ddbe1ee)

I did this but there is an error

This is my repository :  https://github.com/Answere1/achieveyou-next-intl

![Снимок экрана 2024-06-03 205732](https://github.com/amannn/next-intl/assets/153187097/18e4eac7-2e69-40dd-937e-e6945a09c2fc)

**Reply by meJohnnyRobert** [reply]
~~Ohh, I think I know what the issue is!~~

~~After looking at your repository's structure, I think your `messages` folder should be inside the `src` folder, basically I think the error simply says that it can't find the folder because it expects it to be in that location but you seem to have the `messages` folder at the root of your project instead.~~

Nevermind, I have tested your repo and for me it seems to work out of the box, even the language switcher seems to work!

<img width="1439" alt="Screenshot at Jun 03 21-27-20" src="https://github.com/amannn/next-intl/assets/115624777/ff02ac38-1e09-455f-99f4-e60ac78f111f">

I think this may actually be some caching discrepancy on your side, I recommend doing the following:

1. Delete the `.next` folder from your project.
2. Restart Your PC/Laptop
3. Open Project Again
4. Run "npm run dev" and the .next folder will be rebuilt
5. Test if it works

**Reply by Answere1** [reply]
I did as you said and the same error comes out. doing the next folder did not fix the error. 
 I also took your first tip and moved the messages folder to src and it worked.

**Reply by meJohnnyRobert** [reply]
Oh wow, I'm glad to hear that it worked!
